tests/it/rules/d_cmake_interop.d

```diff
@@ -0,0 +1,101 @@
+module tests.it.rules.d_cmake_interop;
+
+import tests.it.runtime;
+import tests.utils;
+import reggae.reggae;
+
+string binaryPath(in string binary) {
+    import std.path : buildNormalizedPath;
+    return buildNormalizedPath(ReggaeSandbox.currentTestPath, ".reggae", binary);
+}
+
+@("D and CMake rules interop - shared library")
+unittest {
+    import std.process : executeShell;
+    import std.format : format;
+    import std.path : buildNormalizedPath;
+
+    with(immutable ReggaeSandbox("d_cmake_shared")) {
+        writeFile("reggaefile.d",
```

Comment:
I was wondering where the `reggaefile.d` was, searching for it. Is there a way you could add it to the project directly, instead of generating it here on the fly?

Reply:
I originally also thought this was a good idea but came to realise that the amount of hackery required to support it (that the codebase is still paying for) isn't worth it. If push comes to shove, one can run the offending test and look at the reggaefile then.

Reply:
Done.